edb/server/compiler/compiler.py

```diff
@@ -1898,11 +1899,17 @@ def _compile_ql_query(
         sql_info_prefix = ''
 
     globals = None
+    permissions = None
     if ir.globals:
         globals = [
             (str(glob.global_name), glob.has_present_arg)
             for glob in ir.globals
         ]
+    if isinstance(ir, irast.Statement) and ir.permissions:
+        permissions = [
+            str(permission.global_name)
+            for permission in ir.permissions
+        ]
```

Comment:
I think maybe we should keep the permissions treated as globals until this stage, and split them here?